json_serializable/lib/src/type_helpers/enum_helper.dart

```diff
@@ -15,26 +16,75 @@ class EnumHelper extends TypeHelper {
   @override
   String serialize(
       DartType targetType, String expression, SerializeContext context) {
-    if (!isEnum(targetType)) {
+    var memberContent = _enumValueMapFromType(targetType);
+
+    if (memberContent == null) {
       return null;
     }
 
-    var nullableLiteral = context.nullable ? '?' : '';
+    context.addMember(memberContent);
 
-    return '$expression$nullableLiteral.toString()'
-        "$nullableLiteral.split('.')$nullableLiteral.last";
+    return '${_constMapName(targetType)}[$expression]';
   }
 
   @override
   String deserialize(
       DartType targetType, String expression, DeserializeContext context) {
-    if (!isEnum(targetType)) {
+    var memberContent = _enumValueMapFromType(targetType);
+
+    if (memberContent == null) {
       return null;
     }
 
+    context.addMember(_enumDecodeHelper);
```

Comment:
So we will actually be adding one of these each time this helper is called right? Is that once per field using the enum or more than that?

Reply:
once per field using the enum – but then they are collapsed.

The storage is a set, so it's only recorded once.

Reply:
kk